packages/api-explorer/src/components/ExploreType/ExploreProperty.tsx

```diff
@@ -0,0 +1,262 @@
+ * @constructor
+ */
+export const ExplorePropertyDescription: FC<ExplorePropertyProps> = ({
+  property,
+}) => {
+  if (!property.description) return <></>
+  return <ReactMarkdown source={property.description} />
+}
+
+/**
+ * Show the details of the property
+ * @param property
+ * @constructor
+ */
+export const ExplorePropertyDetail: FC<ExplorePropertyProps> = ({
+  property,
+}) => (
+  <Space>
+    <FlexItem width="10rem">
+      <ExploreTypeLink type={property.type} />
+    </FlexItem>
+    <FlexItem width="5rem">
+      <ExplorePropertyRequired property={property} />
+      <ExplorePropertyReadOnly property={property} />
+      <ExplorePropertyDeprecated property={property} />
+    </FlexItem>
+    <FlexItem width="30rem">
+      <ExplorePropertyDescription property={property} />
+    </FlexItem>
+  </Space>
+)
+
+/**
+ * Render a potentially complex property as a non-expanding node
+ * @param property to display
+ * @constructor
+ */
+export const ExplorePropertyNode: FC<ExplorePropertyProps> = ({ property }) => {
+  const legend = typeIcon(property.type)
+  return (
+    <TreeItem
+      {...legend}
+      detail={<ExplorePropertyDetail property={property} />}
+    >
+      {property.jsonName}
+    </TreeItem>
+  )
+}
+
+/**
+ * Render the Tree or TreeItem for this property
+ * @param property to display
+ * @param level current nesting level
+ * @param maxDepth maximum depth to expand
+ * @param openAll expands entire tree if true
+ * @constructor
+ */
+export const ExploreProperty: FC<ExplorePropertyProps> = ({
+  property,
+  level = 0,
+  maxDepth = -1,
+  openAll = false,
+}) => {
+  const picked = pickType(property.type)
+  if (!picked.intrinsic) {
+    return (
+      <ExplorePropertyType
+        property={property}
+        open={false}
+        level={level + 1}
+        maxDepth={maxDepth}
+        openAll={openAll}
+      />
+    )
+  }
+  return <ExplorePropertyNode property={property} />
+}
+
+interface ExplorePropertyTypeProps extends ExplorePropertyProps {
+  /** Open the node display immediately? */
+  open?: boolean
+}
+
+export const ExplorePropertyType: FC<ExplorePropertyTypeProps> = ({
+  property,
+  open = true,
+  level = 0,
+  maxDepth = -1,
+  openAll = false,
+}) => {
+  const type = property.type
+  const props = pickTypeProps(type)
+  const nest = expandable(level, maxDepth)
+  const legend = typeIcon(type)
+  if (!nest) {
+    return <ExplorePropertyNode property={property} />
+  }
+  return (
+    <Tree
+      border
+      label={`${property.jsonName}`}
+      icon={legend.icon}
+      defaultOpen={open || openAll}
+      detail={<ExplorePropertyDetail property={property} />}
+    >
+      {nest &&
+        Object.values(props).map((property) => (
+          <ExploreProperty
+            key={property.fullName}
+            property={property}
+            level={level + 1}
+            maxDepth={maxDepth}
+            openAll={openAll}
+          />
+        ))}
+    </Tree>
+  )
+}
```

Comment:
Another candidate for a the single return. Again, this is just a personal opinion so I'll let you decide whether to keep this as is and if you do, you can probably remove the `nest` operand on L250 since it _should_ be redundant when that branch is executed:
```suggestion
  return (
    <>
      {nest ? (
        <Tree
          border
          label={`${property.jsonName}`}
          icon={legend.icon}
          defaultOpen={open || openAll}
          detail={<ExplorePropertyDetail property={property} />}
        >
          {Object.values(props).map((property) => (
            <ExploreProperty
              key={property.fullName}
              property={property}
              level={level + 1}
              maxDepth={maxDepth}
              openAll={openAll}
            />
          ))}
        </Tree>
      ) : (
        <ExplorePropertyNode property={property} />
      )}
    </>
  )
}
```

Reply:
I definitely do not find this suggestion to be clarifying. I won't be taking it.